okhttp/src/jvmMain/kotlin/okhttp3/internal/http2/Http2Stream.kt

```diff
@@ -282,7 +282,10 @@ class Http2Stream internal constructor(
 
     val open: Boolean
     synchronized(this) {
-      if (!hasResponseHeaders || !inFinished) {
+      if (!hasResponseHeaders ||
```

Comment:
Agree about not using inFinished. Is there a cleanup that should be considered for the hasResponseHeaders?  Should it be false until the first non 100?

Reply:
I don't think it matters in practice.

In my first attempt at this I forgot that receiveHeaders is called by both clients and servers, so I handle either `:status` or `:method`.